ioctl/cmd/did/didgenerate.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package did
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/iotexproject/iotex-core/ioctl/cmd/account"
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/output"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+)
+
+var signer string
+
+// Multi-language support
+var (
+	generateCmdShorts = map[config.Language]string{
+		config.English: "Generate DID document using private key from wallet",
+		config.Chinese: "用钱包中的私钥产生DID document",
+	}
+	generateCmdUses = map[config.Language]string{
+		config.English: "generate [-s SIGNER]",
+		config.Chinese: "generate [-s 签署人]",
+	}
+	flagSignerUsages = map[config.Language]string{
+		config.English: "choose a signing account",
+		config.Chinese: "选择一个签名账户",
+	}
+)
+
+// generateCmd represents the generate command
+var generateCmd = &cobra.Command{
+	Use:   config.TranslateInLang(generateCmdUses, config.UILanguage),
+	Short: config.TranslateInLang(generateCmdShorts, config.UILanguage),
+	Args:  cobra.ExactArgs(0),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		err := generate()
+		return output.PrintError(err)
+	},
+}
+
+func init() {
+	generateCmd.Flags().StringVarP(&signer, "signer", "s", "", config.TranslateInLang(flagSignerUsages, config.UILanguage))
```

Comment:
we can use RegisterWriteCommand(), just like all other commands that has '-s' ?

Reply:
let me check this